src/events/EventsManager.ts

```diff
@@ -0,0 +1,23 @@
+import { IEvent } from "./core";
+
+export default class EventsManager {
```

Comment:
Would it be possible to roll over the functionality of the event tracker into this class? Just for the sake of minimizing the objects we need to be responsible for. I would also be down to hear an argument for why the two objects shouldn't be merged.

Reply:
@cheukyin699 opinions appreciated as well :)

Reply:
I view the events manager as a database for the events, where you interface with it to get the actual event (given some id). It probably won't be good to merge the functionality of the event tracker.

just my 30 centss

Reply:
That sounds reasonable then